### ISSUE TITLE
Don't append "py" on show call hierarchy view when source is a notebook file 

### ISSUE THREAD

**Issue by luabud** [microsoft/pylance-release#5815]
<h3> Environment data </h3>

-   Language Server version: 2024.4.104
-   OS and version: linux x64 (Codespaces) 
-   Python version (and distribution if applicable, e.g. Anaconda): Python 3.10.13, venv 
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: basic

<h3> Code Snippet </h3>

Open a notebook file and add the following:
```python

class Class:
    # Single-line ranges (such as single-line functions) are not foldable.
    def singleLineFunction(self): pass

    def function1(self, name: str, address: str, title: str):
        # Multiline strings can be folded, leaving only the first line visible.
        """Doc string
        Doc string
        """
        pass

Class().function1("", "", "")
```
Then right click on function1 at the last line and trigger "Show Call Hierarchy". Notice how the notebook file shows on the callers view   with `.py` appended to it 

![image](https://github.com/microsoft/pylance-release/assets/45497113/dfc85d1a-c2c4-4d5e-913f-1aaf77047ab2)

**Comment by debonte** [comment]
Type hierarchy also has this issue:
![image](https://github.com/microsoft/pylance-release/assets/12820898/05ab2967-d852-47f7-9ddd-43ef875a5005)

This is a slight improvement over these issues tracking the old behavior. I'll close them:
- https://github.com/microsoft/pylance-release/issues/3611
- https://github.com/microsoft/pylance-release/issues/5192

**Comment by KacieKK** [comment]
This issue has been fixed in prerelease version 2024.5.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20245100-9-May-2024-PreRelease)
